### DISCUSSION TITLE
What is the angle resolution of the Robot Inventor Hub motors?

### DISCUSSION THREAD

**Discussion by pepijndevos** [pybricks/support#1235] [Q&A] [answered]
I'm trying to do precise/slow movements to make a plotter, but I'm finding that the motors turn in what appears to be almost 4 degree increments. When I run `track_angle` in a loop, it seems to make a sort of staircase movement.

Is the only way to solve this to just gear down the motor a lot to the point the step size is acceptable?

Simple demonstrator
```python
from pybricks.hubs import InventorHub
from pybricks.pupdevices import Motor, ColorSensor, UltrasonicSensor
from pybricks.parameters import Button, Color, Direction, Port, Side, Stop
from pybricks.robotics import DriveBase
from pybricks.tools import wait, StopWatch

hub = InventorHub()

m = Motor(Port.B)
m.run(5)
while True:
    print(m.angle())
    wait(100)
```

What's weird is that it's not exactly fixed intervals, there is some noise and variation, but it's pretty rough.

Somewhat unrelated: What do I need to pass to make the motor just consider its initial position zero? To avoid huge jumps at the start of the program.

**Comment by laurensvalk** [comment, marked as answer]
The motors have quite a lot of friction, so there’s a lot of stick slip movement going on at low speeds. Anything below 100 deg/s isn’t going to look great. This is equivalent of 10% speed in the LEGO apps.

To reset the angle to zero, try motor.reset_angle(0).

Depending on the use case, you might find that run_angle or run_target works better than track_target because it also does feedforward control instead of only feedback.

**Reply by pepijndevos** [reply]
Ah okay so they do have more resolution but they get jumpy due to friction.
Is the actual resolution in hardware and software one degree?
I guess either way the solution is the same: gear down the motors so I can run them at higher speeds.

Well, so I'm trying to make a plotter that actually runs simple gcode. This means I basically need to make a ton of small linear movements. Maybe run_target would work if I set the stop action to coast? I want to avoid the motor coming to a full stop after each segment, but it sounds like run_target would give me better control and also would allow me to set the relative speed needed for a certain angle.

**Reply by laurensvalk** [reply]
Yes, one degree.

[Also](https://www.youtube.com/watch?v=fGQu90EPVAM) :smile: :

![image](https://github.com/pybricks/support/assets/12326241/b962dc60-ee14-41a4-9992-25581f6fa169)

**Reply by pepijndevos** [reply]
Oh that's very cool! Do you have any idea about the gear ratios and software?

**Reply by laurensvalk** [reply]
See https://www.eurobricks.com/forum/index.php?/forums/topic/186820-wip-mindstorms-51515-plotter/ and https://www.eurobricks.com/forum/index.php?/forums/topic/189951-moc-mindstorms-51515-plotter/ for some discussion.
